static/app/views/automations/components/automationBuilder.tsx

```diff
@@ -0,0 +1,265 @@
+import {useContext} from 'react';
+import styled from '@emotion/styled';
+
+import {Flex} from 'sentry/components/container/flex';
+import {Button} from 'sentry/components/core/button';
+import SelectField from 'sentry/components/forms/fields/selectField';
+import FormContext from 'sentry/components/forms/formContext';
+import {IconAdd, IconDelete, IconMail} from 'sentry/icons';
+import {t, tct} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {DataConditionGroupLogicType} from 'sentry/types/workflowEngine/dataConditions';
+import {
+  type AutomationBuilderAction,
+  AutomationBuilderContext,
+  automationReducer,
+} from 'sentry/views/automations/components/automationBuilderState';
+import RuleNodeList from 'sentry/views/automations/components/ruleNodeList';
+
+// When to use ANY vs ANY_SHORT_CIRCUIT?
+const TRIGGER_MATCH_OPTIONS = [
+  {value: DataConditionGroupLogicType.ALL, label: t('all')},
+  {value: DataConditionGroupLogicType.ANY, label: t('any')},
+];
+
+const FILTER_MATCH_OPTIONS = [
+  {value: DataConditionGroupLogicType.ALL, label: t('all')},
+  {value: DataConditionGroupLogicType.ANY, label: t('any')},
+  {value: DataConditionGroupLogicType.NONE, label: t('none')},
+];
+
+export default function AutomationBuilder() {
+  const context = useContext(AutomationBuilderContext);
+  const formContext = useContext(FormContext);
+  if (!context) {
+    throw new Error('stop what ur doin');
+  }
+  const {state, setState} = context;
+
+  function dispatch(action: AutomationBuilderAction) {
+    setState(currentState => automationReducer(currentState, action, formContext.form));
+  }
```

Comment:
This is essentially what [`useReducer`](https://react.dev/reference/react/useReducer) does. I'd try to use that if possible, to clean up passing the `setState` callback around.